packages/editor/src/hooks/pattern-partial-syncing.js

```diff
@@ -30,37 +30,41 @@ const {
  */
 const withPartialSyncingControls = createHigherOrderComponent(
 	( BlockEdit ) => ( props ) => {
-		const blockEditingMode = useBlockEditingMode();
-		const isEditingPattern = useSelect(
-			( select ) =>
-				select( editorStore ).getCurrentPostType() ===
-				PATTERN_TYPES.user,
-			[]
-		);
-
-		const shouldShowPartialSyncingControls =
-			props.isSelected &&
-			isEditingPattern &&
-			blockEditingMode === 'default' &&
-			Object.keys( PARTIAL_SYNCING_SUPPORTED_BLOCKS ).includes(
-				props.name
-			);
+		const isSupportedBlock = Object.keys(
+			PARTIAL_SYNCING_SUPPORTED_BLOCKS
+		).includes( props.name );
 
 		return (
 			<>
 				<BlockEdit { ...props } />
-				{ shouldShowPartialSyncingControls && (
-					<PartialSyncingControls { ...props } />
+				{ props.isSelected && isSupportedBlock && (
+					<ControlsWithStoreSubscription { ...props } />
 				) }
 			</>
 		);
 	}
 );
 
-if ( window.__experimentalPatternPartialSyncing ) {
-	addFilter(
-		'editor.BlockEdit',
-		'core/editor/with-partial-syncing-controls',
-		withPartialSyncingControls
+// Split into a separate component to avoid a store subscription
+// on every block.
+function ControlsWithStoreSubscription( props ) {
```

Comment:
This is the only bit that I refactored a little bit, as otherwise it would've added a store subscription for every paragraph, heading, image and button. Now it'll only do so when one of those blocks is selected.

Reply:
> it would've added a store subscription for every paragraph, heading, image and button.

Was that coming from a noticeable performance impact? My impression was that a store subscription should be cheap. But this is still a good refactor!

Reply:
Yep, apparently there have been good performance improvements form making these kind of changes, so this follows the trend: https://github.com/WordPress/gutenberg/pulls?q=is%3Apr+label%3A%22%5BType%5D+Performance%22+is%3Amerged+store+subscription+